tests/Microsoft.Identity.Test.Common/Core/Helpers/AssertException.cs

```diff
@@ -103,30 +103,37 @@ public static TException Throws<TException>(Func<object> testCode, bool allowDer
         }
 
         [DebuggerStepThrough]
-        public static T TaskThrows<T>(Func<Task> testCode)
+        public static T TaskThrows<T>(Func<Task> testCode, bool allowDerived = false)
```

Comment:
This does not work well if the exception is thrown before the first await. In this case, the exception is not packed into an AggregateException.